service/internal/ebs/ebs.go

```diff
@@ -208,25 +208,60 @@ func (e *EBS) ListVolumes(cr infrastructurev1alpha2.AWSCluster, filterFuncs ...f
 		if !IsFiltered(v, filterFuncs) {
 			continue
 		}
-
+		ignoreVolume := false
 		attachments := []VolumeAttachment{}
 
 		if len(v.Attachments) > 0 {
 			for _, a := range v.Attachments {
+
+				if *a.InstanceId != "" {
+					i := &ec2.DescribeInstancesInput{
+						InstanceIds: aws.StringSlice([]string{*a.InstanceId}),
+					}
+
+					o, err := e.client.DescribeInstances(i)
+					if err != nil {
+						return nil, microerror.Mask(err)
+					}
+
+					// if the instance does not have the proper cluster-id tag than ignore the volume
+					if len(o.Reservations) > 0 && len(o.Reservations[0].Instances) > 0 &&
+						!instanceHasTag(o.Reservations[0].Instances[0], key.TagCluster, cr.Labels[key.TagCluster]) {
```

Comment:
Can we make this condition a little bit shorter and more readable? 

```go
hasInstances := len(o.Reservations) > 0 && len(o.Reservations[0].Instances) > 0
hasTags := instanceHasTag(o.Reservations[0].Instances[0], key.TagCluster, cr.Labels[key.TagCluster])

if hasInstances && !hasTags {
    ...
}
```

Reply:
I am also wondering if we need to pay attention to `o.Reservations[0].Instances[1]` and beyond. Are there more instances necessary to check?

Reply:
EBS volume can be only attached to a single instance so We don't need to care, also the request to desciribe instances is via `instance-id` which is unique across AWS so result of that request must be a single instance

Reply:
> hasInstances := len(o.Reservations) > 0 && len(o.Reservations[0].Instances) > 0
hasTags := instanceHasTag(o.Reservations[0].Instances[0], key.TagCluster, cr.Labels[key.TagCluster])
if hasInstances && !hasTags {
    ...
}

this does not work as if there are no instances the `hasTags` expression will panics